src/Octoshift/Handlers/DownloadLogsCommandHandler.cs

```diff
@@ -75,20 +73,17 @@ public async Task Handle(DownloadLogsCommandArgs args)
             _log.LogWarning($"Overwriting {args.MigrationLogFile} due to --overwrite option.");
         }
 
-        var result = await _retryPolicy.RetryOnResult(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), string.Empty,
+        var result = await _retryPolicy.RetryOnResult<(string MigrationLogURL, string MigrationID)>(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), result => string.IsNullOrEmpty(result.MigrationLogURL),
             "Waiting for migration log to populate...");
 
-        if (result.Outcome == OutcomeType.Successful && result.Result is null)
```

Comment:
`GetMigrationLogUrl` now throws an exception in this case, rather than returning `null`. This feels like the natural control flow for this unhappy path.